### PR TITLE
Add `archives` parameter to make use with an ENC or hiera easier

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description

This PR adds a parameter to make generating archive requests from an ENC easier.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -32,6 +32,7 @@
   Optional[String[1]]                       $seven_zip_source   = undef,
   Boolean                                   $aws_cli_install    = false,
   Boolean                                   $gsutil_install     = false,
```
Comment (alexjfisher) [inline manifests/init.pp:34]: Best to default to an empty hash instead of `undef`

File: manifests/init.pp [hunk 2]
```diff
@@ -97,4 +98,10 @@
       }
     }
   }
+
+  $archives.each |$archive_name, $archive_settings| {
+    archive { $archive_name
```
Comment (alexjfisher) [inline manifests/init.pp:103]: missing a `:` at the end of this line.

**Review by alexjfisher — CHANGES_REQUESTED** [review]
Thanks for the PR. I'm not against adding this functionality, but it currently has a syntax error.

File: manifests/init.pp [hunk 3]
```diff
@@ -32,6 +32,7 @@
   Optional[String[1]]                       $seven_zip_source   = undef,
   Boolean                                   $aws_cli_install    = false,
   Boolean                                   $gsutil_install     = false,
+  Optional[Hash]                            $archives           = {},
```
Comment (alexjfisher) [inline manifests/init.pp:35]: The data type can be just `Hash` now.

Comment (jcpunk) [reply manifests/init.pp]: I'll fix an push new.

File: manifests/init.pp [hunk 4]
```diff
@@ -32,6 +32,7 @@
   Optional[String[1]]                       $seven_zip_source   = undef,
   Boolean                                   $aws_cli_install    = false,
   Boolean                                   $gsutil_install     = false,
+  Hash                                      $archives           = {},
```
Comment (alexjfisher) [inline manifests/init.pp:35]: Could you document the parameter using the puppet-strings style as above https://github.com/voxpupuli/puppet-archive/pull/423/files#diff-c6c3a824d185ff01e063011517a1b13a604dc3d7b63d1c67a30451c10df02bcbL26?

I dunno, something like...

```puppet
# @param archives
#   A hash of archive resources that this module should create
```

Bonus points for adding a `@example` to the examples section.

Comment (jcpunk) [reply manifests/init.pp:45]: Done

**Review by alexjfisher — APPROVED** [review]

**Comment by kenyon** [comment]
Isn't this for hiera, not an ENC?

**Comment by alexjfisher** [comment]
> Isn't this for hiera, not an ENC?

either.

**Pull request merged**